README.dev.md

```diff
@@ -79,10 +79,68 @@ During the development cycle, three main supporting branches are used:
+     directly, you must [create the release manually](#manually-create-a-release).
+3. Visit [Actions](https://github.com/DeepRank/deeprank2/actions) tab to check whether everything went as expected.
+   - NOTE: there are two separate jobs in the workflow: "draft_release" and "tidy_workspace". The first creates the draft release on github, while the second merges changes into `dev` and closes the PR.
+     - If "draft_release" fails, then there are likely merge conflicts with `main` that need to be resolved first. No release draft is created and the "tidy_workspace" job does not run.
+     - If "draft_release" is succesfull but "tidy_workspace" fails, then there are likely merge conflicts with `dev` that are not conflicts with `main`. In the latter case, the draft release is created and it is up to the user to decide whether to proceed with finalizing the release or first resolving the conflicts with `dev` (this should never happen if the release branch was `dev`, as the only change will be the version bump).
+     - If both jobs succeed, then the draft release is created and the changes are merged into `dev` without any problems and the associated PR is closed. Also, if the release branch is different from `dev`, then that branch will be deleted from the remote repository.
+4. Navigate to the [Releases](https://github.com/DeepRank/deeprank2/releases) tab and click on the newest draft
+   release that was just generated.
+5. Click on the edit (pencil) icon on the right side of the draft release.
+6. Check/adapt the release notes and make sure that everything is as expected.
+7. Check that "Set as the latest release is checked".
+8. Click green "Publish Release" button to convert the draft to a published release on GitHub.
+   - This will automatically trigger [another GitHub workflow](https://github.com/DeepRank/deeprank2/actions/workflows/release.yml) that will take care of publishing the package on PyPi.
+
+#### Updating the token:
+
+NOTE: the current token (associated to @DaniBodor) allowing to bypass branch protection will expire on 9 July 2025. To update the token do the following:
+
+1. [Create a personal access token](https://github.com/settings/tokens/new) from a GitHub user account with admin
+   priviliges for this repo.
+2. Check all the "repo" boxes and the "workflow" box, set an expiration date, and give the token a note.
+3. Click green "Generate token" button on the bottom
+4. Copy the token immediately, as it will not be visible again later.
+5. Navigate to the [secrets settings](https://github.com/DeepRank/deeprank2/settings/secrets/actions).
+6. Edit the `GH_RELEASE` key giving your access token as the new value.
+
+### Manually create a release:
+
+0. Make sure you have all required developers tools installed `pip install -e .'[test]'`.
+1. Create a `release` branch from `main` and merge the changes into this branch.
```

Comment:
Create the release branch from dev (for the workflow we follow in deeprank2). See the old README.dev.md

Reply:
My idea is that we create a `release` branch from main and then merge any changes (whether `dev` or some hotfix branch) into `release`. Alternatively, `dev` can just be used directly, without creating the `release` branch at all, but i wanted to give unified instructions that don't depend on where the changes come from.
I clarified the instructions for this.

Reply:
In the [branching workflow we are using](https://github.com/DeepRank/deeprank2/blob/main/README.dev.md#branching-workflow) the release branch is created from dev, and hotfix branches from main. I'd prefer to stick to the workflow in the automated action. In general, it's useful to have such an automation but it shouldn't make the process less clear than before.

Reply:
I don't understand your point. Maybe we can have a quick call about it rather than going back and forth on GH.

Reply:
```suggestion
1. Create a `release-` branch from `main` (if there has been an hotfix) or `dev` (regular new production release) and prepare the branch for the release (e.g., removing the unnecessary dev files, fix minor bugs if necessary).
```

I'd remove the two following bullet points:
```
   - Ensure that the `release` branch is ready to be merged back into `main` (e.g., removing the unnecessary files, fix minor bugs if necessary).
   - Normally speaking, `release` should contain the changes from `dev` (see our [development workflow](#branching-workflow)), although in some cases a hotfix may be implemented from a different branch.
 ```

Reply:
Remember this that for some reason doesn't appear in my review below @DaniBodor